### PR TITLE
Replace GetMapTileTexture with GetCurrentMapId

### PR DESCRIPTION
**Pull request by aldudko**
Using GetCurrentMapId should provide a cleaner way
of finding a match between maps and LOST_TREASURE_DATA
as GetCurrentMapId doesn't rely on .dds filenames.

### CODE REVIEW

**Comment by simenon** [comment]
In the past it relied on GetCurrentMapZoneIndex() before i switched over on to the texture names, as they are less likely to change. I am not sure on if the GetCurrentMapId() would be reliable if more zones are added.

**Comment by aldudko** [comment]
There are several exception which need to be hardcoded when using GetMapTileTexture.
For example, https://gitter.im/esoui/esoui?at=5dbc5027f26ea4729d368829
Although it doesn't affect LostTreasure right now, because there are no treasure or survey maps in such unconventionally named zones.

As for GetCurrentMapId:

- GetCurrentMapId was added in 2019: https://forums.elderscrollsonline.com/en/discussion/454530/update-21-api-patch-notes-change-log-pts
- GetMapNumTilesForMapId and GetMapTileTextureForMapId require MapId to be passed as an argument. I.e. other API functions rely on what GetCurrentMapId returns.
- MapTables uses GetCurrentMapId https://github.com/Drakanwulf/MapTables/blob/6a6d5dec13c839154b5e97875d70e4e708a9aa76/MapTables.lua#L113

In other words it looks pretty stable to me.

ESO is going to add a new chapter with new zones soon.
What do you think about waiting for it so that I can confirm that nothing is broken?

**Comment by simenon** [comment]
Not relevant anymore
